File: app/src/main/java/com/gmail/fattazzo/formula1world/view/chart/valueformatter/AlternateChartValueFormatter.java
package com.gmail.fattazzo.formula1world.view.chart.valueformatter;

import com.github.mikephil.charting.data.Entry;
import com.github.mikephil.charting.utils.ViewPortHandler;

/**
 * @author fattazzo
 *         <p/>
 *         date: 07/06/17
 */
public class AlternateChartValueFormatter extends DefaultChartValueFormatter {

    @Override
    public String getFormattedValue(float value, Entry entry, int dataSetIndex, ViewPortHandler viewPortHandler) {

        boolean evenData = true;
        if (entry.getData() instanceof Boolean) {
            evenData = (boolean) entry.getData();
        }

        if (evenData) {
            return super.getFormattedValue(value, entry, dataSetIndex, viewPortHandler);
        }
        return "";
    }
}

File: app/src/main/kotlin/com/gmail/fattazzo/formula1world/view/chart/valueformatter/AlternateChartValueFormatter.kt
/*
 * Project: total-gp-world
 * File: AlternateChartValueFormatter.kt
 *
 * Created by fattazzo
 * Copyright © 2018 Gianluca Fattarsi. All rights reserved.
 *
 * MIT License
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.gmail.fattazzo.formula1world.view.chart.valueformatter

import com.github.mikephil.charting.data.Entry
import com.github.mikephil.charting.utils.ViewPortHandler

/**
 * @author fattazzo
 *
 *
 * date: 07/06/17
 */
class AlternateChartValueFormatter : DefaultChartValueFormatter() {

    override fun getFormattedValue(value: Float, entry: Entry?, dataSetIndex: Int, viewPortHandler: ViewPortHandler?): String {

        var evenData = true
        if (entry!!.data is Boolean) {
            evenData = entry.data as Boolean
        }

        return if (evenData) {
            super.getFormattedValue(value, entry, dataSetIndex, viewPortHandler)
        } else ""
    }
}
